### PR TITLE
ENH: Use check for PR instead of status

### PR DESCRIPTION
**Pull request by pllim**
Follow up of #45 to actually use the `set_check` API.

TODO:
- [x] Is this what we really want?
- [x] Fix tests, if they fail.
- [ ] How to play with this before merge for all the scenarios being handled in the PR checks?

### CODE REVIEW

**Comment by astrofrog** [comment]
I think we do want this, because the text is sometimes too long for the status bar. I'm not sure if you can play with it for *all* the scenarios but I'd definitely recommend trying it out on a test repository just to get a sense for what this will look like. If you set it up for a test repo, I can try and open some pull requests to see how it behaves :)

**Comment by bsipocz** [comment]
What would this mean in practice besides adding the "checks" tab on the PR? Would the way how the statuses currently are reported back, so one always need to click to go to the "checks" tab?

**Comment by astrofrog** [comment]
I'm not 100% sure, so it would be nice to see this in action in a test repo

**Comment by pllim** [comment]
@astrofrog , pretty sure we tested this in Tucson last year. You were sitting next to me. And we even hi-fived.

**Comment by Cadair** [comment]
I would imagine it would work like Azure does.

I am onboard with this plan, given we already removed the auto-comment builder functionality :stuck_out_tongue:

File: baldrick/plugins/github_pull_requests.py [hunk 1]
```diff
@@ -99,7 +99,10 @@ def process_pull_request(repository, number, installation, is_new=False):
     for label in pr_handler.labels:
         if label in skip_labels:
             if skip_fails:
-                pr_handler.set_status('failure', "Skipping checks due to {0} label".format(label), current_app.bot_username)
+                pr_handler.set_check(
```
Comment (Cadair) [inline baldrick/plugins/github_pull_requests.py]: Do we need this in check land?

(I honestly don't know, but do we have the same issue with having to override failed statues with checks?)

File: baldrick/plugins/github_pull_requests.py [hunk 2]
```diff
@@ -148,13 +152,16 @@ def process_pull_request(repository, number, installation, is_new=False):
         if full_context.startswith(current_app.bot_username + ':'):
```
Comment (Cadair) [inline baldrick/plugins/github_pull_requests.py]: This is iterating over `existing_statuses` but setting checks, that seems like it's not going to work correctly.

**Comment by astrofrog** [comment]
Oh yeah I do remember testing it now :) We might have to test it again to address some of @Cadair's comments though

File: baldrick/plugins/github_pull_requests.py (hunk 1 above)
Comment (pllim) [reply baldrick/plugins/github_pull_requests.py]: Currently, the real world usage at `astropy` is something like "skipping checks due to Experimental label" with ❌ , which is a clear reminder that we need to revisit it when it is no longer experimental. But if it just does not show up, then what does that mean?

File: baldrick/plugins/github_pull_requests.py (hunk 2 above)
Comment (pllim) [reply baldrick/plugins/github_pull_requests.py]: Oh yeah, good point. But does it make sense to iterate over `existing_checks`? Hmm... Gotta find time to dig at the API doc.

File: baldrick/plugins/github_pull_requests.py (hunk 1 above)
Comment (pllim) [reply baldrick/plugins/github_pull_requests.py]: I vote keeping this for now and push to testing stage and see what happens.

**Comment by pllim** [comment]
@Cadair 's comment address. How do I test this on a test repo? @astrofrog , I remember you doing some magic with the experimental branch, is that the way to go?

File: baldrick/github/github_api.py [hunk 3]
```diff
@@ -62,7 +62,7 @@ def invalidate_cache(self):
     @property
     def _headers(self):
         if self.installation is None:
-            return None
+            return {}
```
Comment (pllim) [inline baldrick/github/github_api.py]: I need this change to test stuff without installation number.

**Comment by pllim** [comment]
Update: I finally got around to fixing the tests. I think this is ready for a test run on the "experimental" branch.

**Comment by Cadair** [comment]
I just deployed this on my test giles instance: https://github.com/Cadair/testgiles/pull/2

**Comment by pllim** [comment]
How you liking it?

**Comment by Cadair** [comment]
It seems to work, I just pushed it out to Giles proper

**Comment by pllim** [comment]
@astrofrog , I made `astropy-bot` experimental branch in sync with master. Then I opened astropy/astropy-bot-sandbox#1 but no GitHub Checks appear. How long does it take for the experimental branch to deploy? More than a few secs?

**Pull request merged**